src/dags/common/path.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import shutil
+from contextlib import suppress
+from pathlib import Path
+
+from airflow.decorators import task
+
+logger = logging.getLogger(__name__)
+
+
+@task  # type: ignore[misc]
+def mk_dir(path: Path, clean_if_exists: bool = False) -> Path:
+    """Create and optionally clean directory.
+
+    It is okay for the directory to be created to already exist.
+    The parameter ``clean_if_exists`` can be used
+    to specify that the directory should be cleaned
+    of any containing directories and files.
+    If this cleaning attempt results in errors,
+    they will simply be ignored;
+    The cleaning is simply a best effort attempt.
+
+    Args:
+        path: Path of the directory to create
+        clean_if_exists: Whether to clean the directory if it already existed.
+
+    Returns:
+        The path of the directory created (or that already existed).
+    """
+    logger.debug("Creating directory '%s'.", path)
+    path.mkdir(parents=True, exist_ok=True)
```

Comment:
Not tempted to use the stdlib `tempfile` package :-)

Reply:
No, not tempted indeed ;-) But it is a good and valid question that warrants an explanation!

What we call temp dirs in our DAGs are not really temp dirs: 
- the location is always the same for each DAG, namely `SHARED_DIR / DAG_ID`, 
- they are kept around.

I would have used the `tempfile` package if I needed random names for the directories to be created, or needed automatic cleanup (eg within a context manager). However the run time of each DAG surpasses that of what a context manager can provide. Hence the usage of `pathlib`

The usage of `mk_dir` to create the so-called "temp" dirs for our DAGs is basically an assertion. Ensure that the directory we want to work in exist. And optionally clean it before we start using it. It is also the reason I called the task `mk_dir` instead of `mk_tmp_dir`.